modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -717,32 +724,38 @@ private CompletableFuture<Boolean> beforeZoneReplicaStarted(LocalPartitionReplic
         );
     }
 
-    private CompletableFuture<Boolean> onZoneReplicaDestroyed(LocalPartitionReplicaEventParameters parameters) {
+    private CompletableFuture<Boolean> onZoneReplicaStopped(LocalPartitionReplicaEventParameters parameters) {
         if (!enabledColocation()) {
             return falseCompletedFuture();
         }
 
-        return inBusyLockAsync(busyLock, () -> {
-            Set<TableImpl> zoneTables = zoneTables(parameters.zonePartitionId().zoneId());
+        Set<TableImpl> zoneTables = zoneTables(parameters.zonePartitionId().zoneId());
 
-            CompletableFuture<?>[] futures = zoneTables.stream()
-                    .map(table -> {
-                        closePartitionTrackers(table.internalTable(), parameters.zonePartitionId().partitionId());
+        CompletableFuture<?>[] futures = zoneTables.stream()
+                .map(table -> supplyAsync(
```

Comment:
You have a bug here, must be:

```
CompletableFuture<?>[] futures = zoneTables.stream()
        .map(table -> supplyAsync(() -> tableStopFuture(table), ioExecutor).thenCompose(identity()))
        .toArray(CompletableFuture[]::new);
```

Reply:
Why? `map -> supplyAsync -> tableStopFuture -> CompletableFuture<Void>`

Reply:
Because the return type of `supplyAsync(() -> tableStopFuture(table))` is `CompletableFuture<CompletableFuture<Void>>`

Reply:
Ouch, my bad. Fixed